packages/@divvi/mobile/src/swap/UnfavorableRateBottomSheet.tsx

```diff
@@ -38,21 +43,112 @@ export default function UnfavorableRateBottomSheet({
           type={BtnTypes.PRIMARY}
           size={BtnSizes.FULL}
           text={t('swapUnfavorableRateBottomSheet.cancel')}
+          touchableStyle={{ height: BUTTON_HEIGHT }}
         />
-        <Button
-          onPress={() => {
+        <SlideButton
+          onComplete={() => {
             forwardedRef.current?.close()
             onConfirm()
           }}
-          type={BtnTypes.SECONDARY}
-          size={BtnSizes.FULL}
-          text={t('swapUnfavorableRateBottomSheet.confirm')}
         />
       </View>
     </BottomSheet>
   )
 }
 
+const SlideButton = ({ onComplete }: { onComplete: () => void }) => {
+  const { t } = useTranslation()
+  const slideThreshold = useRef(1)
+  const pan = useRef(new Animated.Value(0)).current
+  const [completed, setCompleted] = useState(false)
+
+  const onCompleteRef = useRef(onComplete)
+
+  // ensure the latest onComplete is used by the pan responder
+  useEffect(() => {
+    onCompleteRef.current = onComplete
+  }, [onComplete])
+
+  const panResponder = useRef(
+    PanResponder.create({
+      onMoveShouldSetPanResponder: () => !completed,
+      onPanResponderMove: Animated.event([null, { dx: pan }], { useNativeDriver: false }),
+      onPanResponderRelease: (_, gestureState) => {
+        if (gestureState.dx > slideThreshold.current) {
+          Animated.timing(pan, {
+            toValue: slideThreshold.current,
+            duration: 50,
+            useNativeDriver: false,
+          }).start(() => {
+            setCompleted(true)
+            // delay the onComplete to allow the user to see the checkmark
+            setTimeout(onCompleteRef.current, 500)
+            // reset after a second to allow the user to slide again in case the
+            // transaction submission fails
+            setTimeout(() => {
+              setCompleted(false)
+              pan.setValue(0)
+            }, 1000)
+          })
+        } else {
+          Animated.spring(pan, {
+            toValue: 0,
+            useNativeDriver: false,
+          }).start()
+        }
+      },
+    })
+  ).current
+
+  return (
+    <View
+      style={[
+        styles.slideButtonContainer,
+        completed
+          ? { backgroundColor: Colors.buttonSecondaryBackground }
+          : { backgroundColor: Colors.buttonTertiaryBackground },
+      ]}
+      onLayout={(event) => {
+        const { width } = event.nativeEvent.layout
+        // threshold is the left edge of the slider, so we need to subtract the
+        // width of the slider which is same as the height of the button
+        slideThreshold.current = width - BUTTON_HEIGHT
```

Comment:
I don't fully follow this, shouldn't it be `with - sliderWidth` (40)?

Reply:
I'm guessing that it's probably easier to access `BUTTON_HEIGHT` here than the width of the slider component itself? But agreed that it's a little confusing and only happens to work because the slider is circular.

Reply:
It was width of the slider + 2 * margin, which turned out to be the button height as well. Adjusted this